vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
+        return None
+    item = json_object
+    for attribute in attributes:
+        if not attribute in item:
+            LOGGER.error(f"Missing attribute {attribute} in {item}")
+            return None
+        item = item[attribute]
+    return item
+
+
 class GitHubAPIImporter(Importer):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    spdx_license_expression = "CC-BY-4.0"
+    token = os.environ["GH_TOKEN"]
+    endpoint = "https://api.github.com/graphql"
+
+    ecosystems = [
+        "MAVEN",
+        "NUGET",
+        "COMPOSER",
+        "PIP",
+        "RUBYGEMS",
+        "GO",
+    ]
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        headers = {"Authorization": f"token {self.token}"}
+        end_cursor_exp = ""
+        advisories = []
+        for ecosystem in self.ecosystems:
+            while True:
+                query_json = {"query": QUERY % (ecosystem, end_cursor_exp)}
+                resp = requests.post(self.endpoint, headers=headers, json=query_json).json()
+                if resp.get("message") == "Bad credentials":
+                    raise GitHubTokenError("Invalid GitHub token")
+                advisories.extend(process_response(resp, ecosystem=ecosystem))
+                page_info = get_item(resp, "data", "securityVulnerabilities", "pageInfo")
+                if not get_item(page_info, "hasNextPage"):
+                    break
+                end_cursor = get_item(page_info, "endCursor")
+                if end_cursor:
+                    end_cursor = f'"{end_cursor}"'
+                    end_cursor_exp = f"after: {end_cursor}"
+        return advisories
+
+
+def get_ecosystem_from_type(type: str):
+    return PKG_TYPE_ECOSYSTEM_MAPPING.get(type)
+
+
+def get_type_from_ecosystem(ecosystem: str):
+    return ECOSYSTEM_PKG_TYPE_MAPPING.get(ecosystem)
+
+
+def extract_references(reference_data: List):
+    """
+    Yield `reference` by itearting over `reference_data`
+    """
+    for ref in reference_data:
+        url = ref["url"]
+        if not isinstance(url, str):
+            continue
+        if "GHSA-" in url.upper():
+            reference = Reference(url=url, reference_id=url.split("/")[-1])
+        else:
+            reference = Reference(url=url)
+        yield reference
+
+
+def process_name(ecosystem: str, pkg_name: str) -> Optional[Tuple[Optional[str], str]]:
+    """
+    Return `namespace` and `name` by processing `ecosystem` and `pkg_name`
+    """
+    if ecosystem == "MAVEN":
+        artifact_comps = pkg_name.split(":")
+        if len(artifact_comps) != 2:
+            return
+        ns, name = artifact_comps
+        return ns, name
+
+    if ecosystem == "COMPOSER":
         try:
-            self.gh_token = os.environ["GH_TOKEN"]
-        except KeyError:
-            raise GitHubTokenError("Environment variable GH_TOKEN is missing")
+            vendor, name = pkg_name.split("/")
```

Comment:
You could also use `pkg_name.partition("/")` that will never fail and always return three values.